core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -172,15 +172,15 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
 
     @Override
```

Comment:
Oh, I see what you mean. Yes, the strange thing here is that inner queries can be `null` when they are coming from the parser, thats why we need the setters/constructors to allow for `null`. I think those checks are there to prevent NPEs later in the doXContent, because we can't render them otherwise. I saw we don't do these checks in other builders, maybe we should? Otherwise we can choose to not render out these fields, but this will lead to parser errors later (there we only check the existence of this field name, not that the query contained actually parsers to a valid query).
If we remove the checks here we simply get NPEs when rendering, like we would get currently in other builders like And/ConstantScore etc... Will think about this, if you have thoughts let me know.